### PR TITLE
ENH: update auth process to stay compliant

### PR DESCRIPTION
**Pull request by deniederhut**
We had been using a simple account name and password to log in
directly to Gmail's smtp server, but Google no longer supports this
as of today https://support.google.com/accounts/answer/6010255?hl=en

The mail script has been rewritten to use an application-based,
assumed role OAuth workflow.

### CODE REVIEW

**Comment by scoobies** [comment]
Hi!, I'm @scoobies, a friendly bot.

Type ```@scoobies help``` to discover how I can help you.

File: publisher/mail/_mailer.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
 import attr
+import base64
 from email.message import EmailMessage
-import smtplib
-import getpass
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build, Resource
+from googleapiclient.errors import HttpError
```
Comment (mepa) [inline publisher/mail/_mailer.py:7]: These google packages seem likely to be backward compatible over time. However, if convenient, adding a requirements.txt of packages with version numbers may help a future somebody set up a working environment for sending emails.

File: publisher/mail/mail_review_final.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+def main(data_path: str, send: bool = False):
+    scipy_proc = options.cfg2dict(proc_conf)
+    proceedings = scipy_proc["proceedings"]
+
+    template = "reviewer-final.txt"
+
+    # we make this table by hand in google sheets when the committee divvies
+    # up reviewers among papers -- the format we have been using is something like
+    # | paper title | PR url | reviewer one | reviewer two | ... | reviewer n |
+    # where each reviewer field is like "firstname lastname <email@provider>"
```
Comment (mepa) [inline publisher/mail/mail_review_final.py:25]: Great context - thanks for adding these comments

**Review by mepa — APPROVED** [review]
LGTM. Thanks for updating this and sorry for not taking a look earlier.

File: publisher/mail/_mailer.py (hunk 1 above)
Comment (deniederhut) [reply publisher/mail/_mailer.py:7]: I couldn't recover the versions I was using -- different year, different laptop, etc. But I added a reqfile with at least the top level dependencies.

**Pull request merged**